Corpus: Fix titles property

djukicn opened the pull request:
##### Issue
Currently, every time the `titles` property is accessed, the `tolist` method is called on numpy array. This operation can be relatively slow for large corpora (0.7s for friends-transcripts) and can slow down widgets that access the property many times (with this PR, the Concordance widget is approximately 500 times faster on several examples I tested it on so this partially fixes #522).


##### Description of changes
Numpy array `_titles` is converted to list only once  (inside `_set_unique_titles`) instead of every time the property is accessed.


##### Includes
- [X] Code changes
- [ ] Tests
- [ ] Documentation

PrimozGodec commented:
Good catch. I do not like the idea of introducing a new attribute (corpus have too many of them already). I have been thinking if we just return titles as `np.ndarray` and check if widgets that access this property work correctly. As I quickly checked the code I found out that it should be ok.

djukicn commented:
I thought so too but I haven't just removed the `tolist()` because I wasn't sure if something would fail because of that. Now I checked all the widgets that use titles, fixed several tests and added one conversion to Corpus Viewer. It should be ok now.

PrimozGodec commented:
@djukicn Thank you for that. It looks good now. :)
